### PR TITLE
[2020-mnt] DF: add min_samples_split, min_weight_fraction_leaf, min_impurity_decrease, max_leaf_nodes parameters

### PR DESCRIPTION
**Pull request by Pahandrovich**
This is the cherry-pick from PR #533

### CODE REVIEW

**Comment by Pahandrovich** [comment]
https://ecmd.inn.intel.com/commander/link/jobDetails/jobs/eaf1fdfb-302e-f152-aea9-8cdcd4b723a1?

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/agorshk-update_df_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: run
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
nightly
https://ecmd.inn.intel.com/commander/link/jobDetails/jobs/eaf2d0c9-16de-f1f8-8329-8cdcd4b723a1?

File: algorithms/kernel/dtrees/forest/df_training_parameter.cpp [hunk 1]
```diff
@@ -35,11 +35,38 @@ namespace decision_forest
...
+} // namespace interface2
+Status checkImpl(const decision_forest::training::interface2::Parameter & prm)
 {
     DAAL_CHECK_EX(prm.nTrees, ErrorIncorrectParameter, ParameterName, nTreesStr());
     DAAL_CHECK_EX(prm.minObservationsInLeafNode, ErrorIncorrectParameter, ParameterName, minObservationsInLeafNodeStr());
+    DAAL_CHECK_EX(prm.minObservationsInSplitNode, ErrorIncorrectParameter, ParameterName, minObservationsInSplitNodeStr());
+    DAAL_CHECK_EX((prm.minWeightFractionInLeafNode >= 0.0) && (prm.minWeightFractionInLeafNode <= 0.5), ErrorIncorrectParameter, ParameterName,
+                  minWeightFractionInLeafNodeStr());
+    DAAL_CHECK_EX(prm.minImpurityDecreaseInSplitNode >= 0.0, ErrorIncorrectParameter, ParameterName, minImpurityDecreaseInSplitNodeStr());
+    DAAL_CHECK_EX((prm.maxLeafNodes == 0) || (prm.maxLeafNodes > 1), ErrorIncorrectParameter, ParameterName, maxLeafNodesStr());
```
Comment (agorshk) [inline algorithms/kernel/dtrees/forest/df_training_parameter.cpp:69]: please remove below, now we support all available maxLeafNodes
DAAL_CHECK_EX((prm.maxLeafNodes == 0) || (prm.maxLeafNodes > 1), ErrorIncorrectParameter, ParameterName, maxLeafNodesStr());

**Review by agorshk — APPROVED** [review]

**Comment by Pahandrovich** [comment]
/intelci: run
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Comment by Pahandrovich** [comment]
/intelci: restart
set infra_branch=dev/pyakovle-df_update_tests_mnt

**Pull request merged**